### DISCUSSION TITLE
bucket replication with existing-objects on is too slow

### DISCUSSION THREAD

**Discussion by majidrajabi** [minio/minio#13107] [Q&A] [answered]
## Expected Behavior

replicate the existing objects of a replicated bucket on the destination bucket in a logical manner
## Current Behavior

replicate the existing data too slow or randomly which we don't know how much time it takes to complete.
## Possible Solution

if this is expected behavior it should report the state of the process of replicating the old data. if not, the speed of the replicating process must be increased. 
## Steps to Reproduce (for bugs)

1. mc admin bucket remote add src/mybucket https://secret:key@myminio.example.com/mybucket --service "replication"
2.  mc replicate add src/mybucket --replicate existing-objects --remote-bucket mybucket  --arn arn:minio:replication::...
3.
4.

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):
minio with docker
docker image : minio/minio:RELEASE.2021-06-17T00-10-46Z
4 data volume
version enabled on both src and dst bucket

**Comment by klauspost** [comment, marked as answer]
@majidrajabi Replication check speed is dependent on a lot of factors. Several factors like the scanner speed, replication workers can be tweaked.

You should have consistent systems at both ends. Replication is not erasure coding and for good performance systems should be able to rely on the remote state to remain consistent. What you describe would be unacceptable in most setups since too many resources would have to be dedicated to consistency checking between clusters.

If you would like us to go through your setup requirements, we offer a [subscription service](https://min.io/pricing) where we can go into details with you.

**Reply by JuryA** [reply]
Now that’s an answer! It gives me the impression of: “So what? Six out of ten files go through without errors, and speed? No one here cares about that!” And then we top it all off with a statement like, “If you have an issue with it, pay up! Otherwise, we don’t care.” No offense...

**Reply by harshavardhana** [reply]
@klauspost meant that infrastructure plays a role—i.e., the internode network, site-to-site link reliability, etc. We already have replicated deployments that are multi-petabyte scaled, deployed, and scaling well. 

The problem is that with toy setups, we can only do a little; inherent infrastructure limits bind us. Architecture guidance requires payment so we can work with you to ensure a successful deployment.

@JuryA, I see you are fixating on the wrong things. I'm sorry to break it to you, but our time is not free.
